Objective-C/CBLDatabase.mm

```diff
@@ -259,6 +259,8 @@ - (BOOL) saveDocument: (CBLMutableDocument *)document
                 if (conflictHandler(document, oldDoc.isDeleted ? nil : oldDoc)) {
                     CBLAssertNotNil(document);
```

Comment:
Why this assertion? The conflictHandler can't change the value of `document`.

Reply:
It was added to make sure, the document is never released or not became dangling pointer. in which case, this will fail instead of other unexpected places.

Reply:
Nothing's changing the value of `document`, so it still has whatever value it originally had. It can't possibly be nil unless it started out that way. (Deallocing the _object_ doesn't affect a _pointer_ to the object.)

(Actually I guess there is the special case of a weak pointer, which gets set to nil if the object is dealloced. But that's not happening here.)

Reply:
done!